invocation.md

```diff
@@ -0,0 +1,1170 @@
+          "iss": "did:key:z6Mkqa4oY9Z5Pf5tUcjLHLUsDjKwMC95HGXdE1j22jkbhz6r",
+          "aud": "did:web:ucan.run",
+          "with": "https://example.com/blog/posts",
+          "do": "crud/create",
+          "input": {
+            "headers": {
+              "content-type": "application/json"
+            },
+            "payload": {
+              "title": "How UCAN Tasks Changed My Life",
+              "body": "This is the story of how one spec changed everything...",
+              "topics": ["authz", "journal"],
+              "draft": true
+            }
+          },
+          "s": {
+            "/": {
+              "bytes:": "5vNn4--uTeGk_vayyPuNTYJ71Yr2nWkc6AkTv1QPWSgetpsu8SHegWoDakPVTdxkWb6nhVKAz6JdpgnjABppC7"
+            }
+          },
+          "prf": [
+            {
+              "/": "bafybeia3tspzaay4gcx3npcczgidbrsutq7yxnag3sfzrmvua6ogqjwy7a"
+            }
+          ]
+        },
+        "ok"
+      ]
+    },
+    "sendEmail": {
+      "<-": [
+        {
+          "iss": "did:key:z6Mkqa4oY9Z5Pf5tUcjLHLUsDjKwMC95HGXdE1j22jkbhz6r",
+          "aud": "did:web:ucan.run",
+          "with": "mailto:akiko@example.com",
+          "do": "msg/send",
+          "input": {
+            "to": ["boris@example.com", "carol@example.com"],
+            "subject": "Coffee",
+            "body": "Hey you two, I'd love to get coffee sometime and talk about UCAN Tasks!"
+          },
+          "s": {
+            "/": {
+              "bytes:": "5vNn4--uTeGk_vayyPuNTYJ71Yr2nWkc6AkTv1QPWSgetpsu8SHegWoDakPVTdxkWb6nhVKAz6JdpgnjABppC7"
+            }
+          },
+          "prf": [
+            {
+              "/": "bafybeia3tspzaay4gcx3npcczgidbrsutq7yxnag3sfzrmvua6ogqjwy7a"
+            }
+          ]
+        },
+        "ok"
+      ]
+    },
+    "sendTextMessage": {
+      "<-": [
+        {
+          "/": "bafybeibwlfwol5bdwj75hdqs3liv6z7dyqwtd2t4ovvgncv4ixkaxlkfle"
+        },
+        "ok"
+      ]
+    }
+  },
+  "s": {
+    "/": {
+      "bytes:": "5vNn4--uTeGk_vayyPuNTYJ71Yr2nWkc6AkTv1QPWSgetpsu8SHegWoDakPVTdxkWb6nhVKAz6JdpgnjABppC7"
+    }
+  }
+}
+```
+
+# 4 Receipt
+
+An Invocation Receipt is an attestation of the Result of an Invocation. A Receipt MUST be signed by the Executor (the `aud` of the associated UCANs) or it's delegate, in which case proof of delegation (of the invoked capability) from Executor to the Signer (the `iss` of the receipt) MUST be provided in `prf`.
+
+**NB: a Receipt this does not guarantee correctness of the result!** The statement's veracity MUST be only understood as an attestation from the executor.
+
+Receipts MUST use the same version as the invocation that they contain.
+
+## 4.1 Schema
+
+```ipldsch
+type Receipt<In, Out> struct {
+  job     &Invocation<In>
+
+  # output of the invocation
+  out     Out
+
+  # Related receipts
+  origin optional &Receipt<In, Any>
+
+  # All the other metadata
+  meta    { String: Any }
+
+  # Principal that issued this receipt
+  iss     Principal
```

Comment:
Can you infer this from the underlying UCAN on the Invocation? What if they don't match?